webapp/channels/src/components/channel_invite_modal/team_invite_banner/team_invite_banner.tsx

```diff
@@ -0,0 +1,407 @@
+        const commaSeparatedUsernames = getCommaSeparatedUsernames(guests);
+        const firstName = guests[0].username;
+        const lastName = guests[guests.length - 1].username;
+        if (guests.length > 10) {
+            return (
+                formatMessage(
+                    {
+                        id: t('channel_invite.invite_team_members.guests.messageOverflow'),
+                        defaultMessage: '{firstUser} and {others} are guest users and need to first be invited to the team before you can add them to the channel. Once they\'ve joined the team, you can add them to this channel.',
+                    },
+                    {
+                        firstUser: (
+                            <Markdown
+                                message={`@${firstName}`}
+                                options={{
+                                    atMentions: true,
+                                    mentionKeys,
+                                    mentionHighlight: false,
+                                    singleline: true,
+                                }}
+                            />
+                        ),
+                        others: (
+                            <SimpleTooltip
+                                id={'usernames-overflow'}
+                                content={commaSeparatedUsernames.replace(`@${firstName}, `, '')}
+                            >
+                                <span
+                                    className='add-others-link'
+                                >
+                                    <FormattedMessage
+                                        id='channel_invite.invite_team_members.messageOthers'
+                                        defaultMessage='{count} others'
+                                        values={{
+                                            count: guests.length - 1,
+                                        }}
+                                    />
+                                </span>
+                            </SimpleTooltip>
+                        ),
+                    },
+                )
+            );
+        }
+
+        const message: string = formatMessage(
+            {
+                id: t('channel_invite.invite_team_members.guests.message'),
+                defaultMessage: '{count, plural, =1 {{firstUser} is a guest user and needs} other {{users} and {lastUser} are guest users and need}} to first be invited to the team before you can add them to the channel. Once they\'ve joined the team, you can add them to this channel.',
+            },
+            {
+                count: guests.length,
+                users: commaSeparatedUsernames.replace(`, @${lastName}`, ''),
+                firstUser: `@${firstName}`,
+                lastUser: `@${lastName}`,
+                team: team.display_name,
+            },
+        );
+
+        return (
+            <Markdown
```

Comment:
I don't think we are supposed to be putting translations though markdown formatting.